testing/index.rst

```diff
@@ -0,0 +1,12 @@
+====================================
+ Testing and Continuous Integration
+====================================
```

Comment:
```suggestion
===================
Testing & Buildbots
===================
```

I was tricked by the original title of the CI page; this is actually specifically about buildbots, not CI in general (or specifically on cloud CI providers, as most OSS developers would assume when reading this). This change another benefit—it makes the title short enough to fit on one line, which means all the category titles do so (which is really nice).

Additionally, use `&` which is shorter and consistent with what is used in the page titles.